corehq/sql_accessors/sql_templates/get_location_fixture_ids.sql

```diff
@@ -75,40 +135,42 @@ BEGIN
                 loc."parent_id",
                 CASE
                     WHEN (
-                        loc_type."expand_from_root" = FALSE
-                        AND loc_type."expand_from" IS NOT NULL
-                        AND NOT (
-                            loc_type."expand_from" = loc."location_type_id"
-                            AND loc_type."expand_from" IS NOT NULL
-                        )
+                        -- if expand_from is set and not the current location type
+                        -- it will be one of this location's ancestors
+                        loc_type."expand_from" IS NOT NULL
+                        AND loc_type."expand_from_root" = FALSE
+                        AND loc_type."expand_from" <> loc."location_type_id"
```

Comment:
this used to say

    AND NOT (
        loc_type."expand_from" = loc."location_type_id"
        AND loc_type."expand_from" IS NOT NULL
    )

and now it says

    AND loc_type."expand_from" <> loc."location_type_id"

these two statements will behave differently if expand_from is NULL because the result of any comparison with NULL is always False. so if expand_from is NULL then this entire expression will evaluate to False now whereas before the entire expression would always evaluate to True when expand_from is NULL

Reply:
Good eye. This condition will never be matched when `expand_from` is NULL because of line 140 (three lines above). The previous form was logically redundant and I only noticed that when converting it from Python to SQL.